pkg/networkserver/downlink.go

```diff
@@ -184,83 +197,111 @@ func (ns *NetworkServer) generateDownlink(ctx context.Context, dev *ttnpb.EndDev
 			logger.Debug("Need downlink for confirmed uplink")
 		}
 	}
-	if !needsDownlink &&
-		len(cmdBuf) == 0 &&
-		len(dev.QueuedApplicationDownlinks) == 0 {
-		return nil, errNoDownlink
-	}
-
-	pld := &ttnpb.MACPayload{
-		FHDR: ttnpb.FHDR{
-			DevAddr: dev.Session.DevAddr,
-			FCtrl: ttnpb.FCtrl{
-				Ack: up != nil && up.Payload.MHDR.MType == ttnpb.MType_CONFIRMED_UP,
-				ADR: deviceUseADR(dev, ns.defaultMACSettings),
-			},
-			FCnt: dev.Session.LastNFCntDown + 1,
-		},
-	}
-	logger = logger.WithField("ack", pld.FHDR.FCtrl.Ack)
 
-	var appDown *ttnpb.ApplicationDownlink
-	if len(cmdBuf) <= fOptsCapacity && len(dev.QueuedApplicationDownlinks) > 0 {
-		down := dev.QueuedApplicationDownlinks[0]
+	var start int
+	for _, down := range dev.QueuedApplicationDownlinks {
 		if len(down.FRMPayload) > int(maxDownLen) {
-			logger.Warn("Application downlink present, but the payload is too long, inform Application Server")
-			go func() {
-				ok, err := ns.handleASUplink(ctx, dev.EndDeviceIdentifiers.ApplicationIdentifiers, &ttnpb.ApplicationUp{
+			logger.Debug("Drop downlink with payload exceeding band regulations")
```

Comment:
Add some log context on two levels;

1. On the message level, i.e. in the loop (like frame counter)
2. On the particular error level, i.e. in this case `frm_payload_size`